cccatalog-api/cccatalog/api/controllers/search_controller.py

```diff
@@ -3,28 +3,74 @@
 from elasticsearch.exceptions import AuthenticationException, \
     AuthorizationException, NotFoundError
 from elasticsearch_dsl import Q, Search, connections
-from elasticsearch_dsl.response import Response
+from elasticsearch_dsl.response import Response, Hit
 from cccatalog import settings
 from django.core.cache import cache
 from cccatalog.api.models import ContentProvider
 from rest_framework import serializers
 import logging as log
+from cccatalog.settings import THUMBNAIL_PROXY_URL, PROXY_THUMBS, PROXY_ALL
+from cccatalog.api.utils.validate_images import validate_images
+from cccatalog.api.utils.dead_link_mask import get_query_mask, get_query_hash
+from rest_framework.reverse import reverse
+from itertools import accumulate
+from typing import Tuple, List, Optional
+from math import ceil
 
 ELASTICSEARCH_MAX_RESULT_WINDOW = 10000
 CACHE_TIMEOUT = 10
+DEAD_LINK_RATIO = 1 / 2
 
 
-def _paginate_search(s: Search, page_size: int, page: int):
+def _paginate_with_dead_link_mask(s: Search, page_size: int,
+                                  page: int) -> Tuple[int, int]:
+    '''
```

Comment:
Use triple double quotes instead of single quotes for consistency with other docstrings in the project

Reply:
Sure.